huaweicloud/services/eg/resource_huaweicloud_eg_event_subscription.go

```diff
@@ -0,0 +1,287 @@
+		}
+		if transformRaw := target["transform"].(string); transformRaw != "" {
+			element["transform"] = unmarshalEventSubscriptionParamsters("transform of event target", transformRaw)
+		}
+		result[i] = element
+	}
+	return result
+}
+
+func resourceEventSubscriptionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+		opts   = subscriptions.CreateOpts{
+			ChannelId:   d.Get("channel_id").(string),
+			Name:        d.Get("name").(string),
+			Description: d.Get("description").(string),
+			Sources:     buildEventSourcesOpts(d.Get("sources").([]interface{})),
+			Targets:     buildEventTargetsOpts(d.Get("targets").([]interface{})),
+		}
+	)
+
+	client, err := cfg.EgV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating EG v1 client: %s", err)
+	}
+
+	resp, err := subscriptions.Create(client, opts)
+	if err != nil {
+		return diag.Errorf("error creating EG subscription: %s", err)
+	}
+	d.SetId(resp.ID)
+
+	return resourceEventSubscriptionRead(ctx, d, meta)
+}
+
+func resourceEventSubscriptionRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg            = meta.(*config.Config)
+		region         = cfg.GetRegion(d)
+		subscriptionId = d.Id()
+	)
+	client, err := cfg.EgV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating EG v1 client: %s", err)
+	}
+
+	resp, err := subscriptions.Get(client, subscriptionId)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "EG subscription")
+	}
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("channel_id", resp.ChannelId),
+		d.Set("name", resp.Name),
+		d.Set("description", resp.Description),
+		d.Set("status", resp.Description),
+		d.Set("created_at", resp.CreatedTime),
+		d.Set("updated_at", resp.UpdatedTime),
+	)
+	if err := mErr.ErrorOrNil(); err != nil {
+		return diag.Errorf("error saving EG event subscription (%s) fields: %s", subscriptionId, err)
+	}
+	return nil
+}
+
+func resourceEventSubscriptionUpdate(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg            = meta.(*config.Config)
+		region         = cfg.GetRegion(d)
+		subscriptionId = d.Id()
+		opts           = subscriptions.UpdateOpts{
+			SubscriptionId: subscriptionId,
+			Description:    utils.String(d.Get("description").(string)),
+			Sources:        buildEventSourcesOpts(d.Get("sources").([]interface{})),
+			Targets:        buildEventTargetsOpts(d.Get("targets").([]interface{})),
+		}
+	)
+	client, err := cfg.EgV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating EG v1 client: %s", err)
+	}
+
+	_, err = subscriptions.Update(client, opts)
+	if err != nil {
+		return diag.Errorf("error updating EG subscription (%s): %s", subscriptionId, err)
+	}
+	return nil
```

Comment:
When doing update operation, It is better to run `Read` function to reSet value.
```suggestion
	return resourceEventSubscriptionRead(ctx, d, meta)
```

Reply:
fixed.